tests-e2e/ocw-ci-test-course/video.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { test, expect } from "@playwright/test"
+import { CoursePage } from "../util"
+
+test("Verify that the 'Download video' and 'Download transcript' links are keyboard navigable and have the correct download URLs", async ({
+  page
+}) => {
+  const course = new CoursePage(page, "course")
+  await course.goto("resources/ocw_test_course_mit8_01f16_l01v01_360p")
+  const downloadLinks = [
+    "https://live-qa.ocw.mit.edu/courses/123-ocw-ci-test-course-fall-2022/ocw_test_course_mit8_01f16_l01v01_360p_360p_16_9.mp4",
+    "https://live-qa.ocw.mit.edu"
+  ]
+  const downloadButton = page.getByRole("button", {
+    name: `Download button`
+  })
+  await downloadButton.focus()
+  await page.keyboard.press("Enter")
+
+  const videoDownloadLink = page.getByRole("link", { name: "Download video" })
+  const transcriptDownloadLink = page.getByRole("link", {
+    name: "Download transcript"
+  })
+  const downloadLinksArr = [videoDownloadLink, transcriptDownloadLink]
+
+  for (let i = 0; i < 2; i++) {
+    await expect(downloadLinksArr[i]).toBeVisible()
+    await page.keyboard.press("Tab")
+    const hrefAttribute = await page.locator(":focus").getAttribute("href")
+    expect(hrefAttribute).toBe(downloadLinks[i])
+  }
+})
+
+test("Embed video redirects to video page using keyboard navigation", async ({
+  page
+}) => {
+  const course = new CoursePage(page, "course")
+  await course.goto("pages/video-series-overview/")
+  const videoRedirectLink = page.getByRole("link", {
+    name: "View video page"
+  })
+  await videoRedirectLink.focus()
+  page.keyboard.press("Enter")
+  await course.page.waitForNavigation()
+  expect(course.page.url()).toContain(
+    "resources/ocw_test_course_mit8_01f16_l01v01_360p"
+  )
+})
+test("Video tabs content (links) are keyoard navigable", async ({ page }) => {
+  const tabDict: { [key: string]: { title: string; redirectPath: string } } = {
+    "related-resource": {
+      title:        "Related Resources",
+      redirectPath: "courses/ocw-ci-test-course/resources/example_pdf/"
+    },
+    "optional-tab": {
+      title:        "Optional Tab",
+      redirectPath: "courses/ocw-ci-test-course/resources/example_notes/"
+    }
+  }
+  for (const tabClass in tabDict) {
+    if (tabClass) {
```

Comment:
**Requested Change:** In general, I would avoid using the `in` operator, especially for iteration. `for (const key in dict)` will iterate over some keys included in the prototype, which is almost never what one wants. I think ESLint raised a warning, which is why `if (tabClass)` got added here.

If you need to async-iterate over a dictionary, I would use `for (const [key, value] of whateverDict) {...}`.

In this case, you don't need the key, just the values. So I would recommend using an array:

```ts
const tabs = [
  {
    title:        "Related Resources",
    url: "courses/ocw-ci-test-course/resources/example_pdf/"
  },
  {
    title:        "Optional Tab",
    url: "courses/ocw-ci-test-course/resources/example_notes/"
  }
]

for (const tab of tabs) {
  // do your stuff
} 
```

Reply:
Done

Reply:
And thanks for the explanation!